Test/Optimizer/test_sr_onthefly_logic.py

```diff
@@ -62,21 +63,14 @@ def random_split_like_tree(rng_key, target=None, treedef=None):
 def tree_random_normal_like(rng_key, target):
     keys_tree = random_split_like_tree(rng_key, target)
     return jax.tree_multimap(
-        lambda l, k: jax.random.normal(k, l.shape, l.dtype), target, keys_tree
+        # TODO remove astype once its fixed in jax
```

Comment:
I wrote that code too... 
Can you file a fix upstream? They don't really care about complex distributions

```suggestion
        # TODO remove astype once its fixed in Jax#6052
```